packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -236,8 +267,31 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
 
   /// Builder for iOS-style copy/paste text selection toolbar.
   @override
-  Widget buildToolbar(BuildContext context, Rect globalEditableRegion, Offset position, TextSelectionDelegate delegate) {
+  Widget buildToolbar(
+      BuildContext context,
+      Rect globalEditableRegion,
+      TextSelectionPoint leftTextSelectionPoint,
+      TextSelectionPoint rightTextSelectionPoint,
+      TextSelectionDelegate delegate) {
     assert(debugCheckHasMediaQuery(context));
+
+    // If the distance from the top is less than a certain value,
+    // the toolbar should be displayed below the input box.
+    // FIX https://github.com/flutter/flutter/issues/29808
```

Comment:
Instead of referring to the bug, we can just say here that it's to avoid showing the toolbar under MediaQuery padding obstructions.